onnxscript/rewriter/ort_fusions/mha.py

```diff
@@ -69,7 +90,11 @@ def pattern(
             _outputs=["key_BSHDh"],
         )
         # Transpose from (B, S, H, D/H) to (B, H, S, D/H)
-        key_BHSDh = op.Transpose(key_BSHDh, perm=[0, 2, 1, 3])
+        # TODO: Fix condition
+        if not self._is_cross_attention and self._has_past_present:
+            key_BHSDh = op.Transpose(key_BSHDh, perm=[0, 2, 1, 3])
+        else:
+            key_BHSDh = op.Transpose(key_BSHDh, perm=[0, 2, 3, 1])
```

Comment:
Minor forward-looking suggestion: such variations can be handled in the check-condition, by allowing perm to be a pattern variable and using `perm=perm` in the pattern. Mentioning this since the restricted disjunction will not allow two such variants with same op.

Reply:
But this is fine for current PR ... this just FYI